image/image.go

```diff
@@ -795,13 +795,14 @@ func (img *Image) IsEncrypted() bool {
 	enc := false
 	if img.Header.Flags&IMAGE_F_ENCRYPTED != 0 {
 		enc = true
-	} else {
-		for _, tlv := range img.ProtTlvs {
-			if tlv.Header.Type&IMAGE_TLV_SECRET_ID_LEGACY != 0 || tlv.Header.Type&IMAGE_TLV_SECRET_ID != 0 {
-				enc = true
-			}
+	} else if res, _ := img.FindAllUniqueTlv(IMAGE_TLV_SECRET_ID_LEGACY); res != nil {
```

Comment:
I think all of the If / else need to be inside this else {}.  In addition, I'd split the assignment and check of the res into two lines.
res, _ := img.FindAllUniqueTlv(XXXXXXX);
if (res != nil } {.....}
etc.